### PR TITLE
[chore] return config from server start methods

### PR DESCRIPTION
**Pull request by benmccann**
We shouldn't read the Kit config in the CLI, but rather in the Vite plugin so that it will work with the `vite` CLI as well. This doesn't move the config reading all the way to the plugin yet, but does remove it from the CLI which is a step in that direction

### CODE REVIEW

File: packages/kit/src/cli.js [hunk 1]
```diff
@@ -61,25 +61,23 @@ prog
...
 				port: address_info.port,
 				host: address_info.address,
-				https: !!(https || server_config.https),
-				open: first && (open || !!server_config.open),
-				base: config.kit.paths.base,
-				loose: server_config.fs.strict === false,
-				allow: server_config.fs.allow
+				https: !!(https || config.server.https),
+				open: first && (open || !!config.server.open),
+				base: config.base,
```
Comment (Rich-Harris) [inline packages/kit/src/cli.js:78]: this will break when we switch to always using `./` with Vite 3

Comment (benmccann) [reply packages/kit/src/cli.js:78]: Why? Because of the leading `.` or something else?

Comment (Rich-Harris) [reply packages/kit/src/cli.js:78]: Because it will open to `localhost:${port}` instead of `localhost:${port}/${config.kit.base}`

Comment (benmccann) [reply packages/kit/src/cli.js:78]: Fixed.

I'm not that familiar with the base path stuff. I was assuming `config.kit.paths.base` would just be passed in to Vite as `config.base`, but it looks we're hardcoding the Vite `config.base` to `'/'`. I switched it to return the SvelteKit config from the server start methods so we can get the correct base path

**Pull request merged**